src/test/java/edu/cornell/kfs/sys/service/impl/AwsSecretServiceImplIntegrationTest.java

```diff
@@ -77,25 +97,79 @@ void testBooleanSetAndGet() {
         assertEquals(expectedNewBoolean, actualNewBoolean);
     }
     
+    @Test
+    void testBooleanSetAndGetWithOutCache() {
+        boolean initialValue = awsSecretServiceImpl.getSingleBooleanFromAwsSecret(SINGLE_BOOLEAN_SECRET_KEY_NAME, false);
+        boolean expectedNewBoolean = !initialValue;
+        awsSecretServiceImpl.updateSecretBoolean(SINGLE_BOOLEAN_SECRET_KEY_NAME, false, expectedNewBoolean);
+        awsSecretServiceImpl.clearCache();
+        
+        waitForAwsSecretUpdateToFullyPost();
+        
+        boolean actualNewBoolean = awsSecretServiceImpl.getSingleBooleanFromAwsSecret(SINGLE_BOOLEAN_SECRET_KEY_NAME, false);
+        
+        assertEquals(expectedNewBoolean, actualNewBoolean);
+    }
+
+    protected void waitForAwsSecretUpdateToFullyPost() {
+        LOG.info("waitForAwsSecretUpdateToFullyPost, waiting 5 seconds");
+        try {
+            TimeUnit.SECONDS.sleep(5);
+        } catch (InterruptedException e) {
+            LOG.error("waitForAwsSecretUpdateToFullyPost. had an error waiting", e);
+            throw new RuntimeException(e);
+        }
+    }
+    
     @Test
     void testPojo() throws JsonMappingException, JsonProcessingException {
+        testPojoWithCache();
+        awsSecretServiceImpl.clearCache();
+        testPojoWithOutCache();
```

Comment:
The previous versions of this PR had separate test cases for the POJO-with-cache and POJO-without-cache tests. Was there a particular reason why you decided to combine them into one test?

Such a setup also got me to think about another issue. This class's wait-for-posting delay is only being called in the middle of the test-without-cache methods, after making an explicit secret update. Since those methods affect the same secrets as their related test-with-cache methods, and since the test-with-cache ones are not waiting for the secret update to post in AWS, then is another wait-for-posting delay needed at the start of the test-without-cache methods (or end of the test-with-cache methods)? That would help avoid using stale secret values at the start of some of the test-without-secret methods, given that they retrieve the existing secrets first.

Reply:
adding the extra wait seems to have avoided the issue that caused me to make this change ..... thanks, good idea